week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -20,34 +20,50 @@ This means the order the songs are played in will be random, but each song will
 
 
 class MusicPlayer {
-    // Add your code here
+    constructor() {
+        this.playing = [];
+        this.MusicPlayer = [];
+        this.index = 0;
+    }
+    add(playing, player) {
+        this.playing.push(playing);
+        this.MusicPlayer.push(player);
+
+    }
+    play() {
+        console.log("Currently playing: " + this.playing[this.index] + " by " + this.MusicPlayer[this.index]);
+
+    }
+    skip() {
+        if (this.index < this.playing.length)
+            this.index++;
```

Comment:
this may throw an error as you're incrementing after the check 
so in array of 5 elements 
if the index is 4, the check will pass but as you're incrementing the index after, it will throw an error